### PR TITLE
Add support for skim binary instead of fzf binary

### PR DESCRIPTION
**Pull request by shenek**
* added a compile time feature `use_skim` which causes that `sk` binary is used instead of `fzf`.
* refactored some code to be more `fzf/skim` neutral.

related issue #227

### CODE REVIEW

File: src/structures/option.rs [hunk 1]
```diff
@@ -37,13 +37,13 @@ pub struct Config {
     #[structopt(long)]
     pub no_preview: bool,
 
-    /// FZF overrides for cheat selection
-    #[structopt(long, env = "NAVI_FZF_OVERRIDES")]
-    pub fzf_overrides: Option<String>,
+    /// finder overrides for cheat selection
+    #[structopt(long, env = "NAVI_FINDER_OVERRIDES")]
```
Comment (denisidoro) [inline src/structures/option.rs]: This is a breaking change :(

I think we should keep the old name even if that doesn't make much sense. At least for now

File: src/structures/option.rs [hunk 2]
```diff
@@ -37,13 +37,13 @@ pub struct Config {
...
+    pub finder_overrides: Option<String>,
 
-    /// FZF overrides for variable selection
-    #[structopt(long, env = "NAVI_FZF_OVERRIDES_VAR")]
-    pub fzf_overrides_var: Option<String>,
+    /// finder overrides for variable selection
+    #[structopt(long, env = "NAVI_FINDER_OVERRIDES_VAR")]
```
Comment (denisidoro) [inline src/structures/option.rs]: Same as above

File: Cargo.toml [hunk 3]
```diff
@@ -14,6 +14,10 @@ license = "Apache-2.0"
 [badges]
 travis-ci = { repository = "denisidoro/navi", branch = "master" }
 
+[features]
+default = []
+use_skim = []
```
Comment (denisidoro) [inline Cargo.toml]: Where is the skim dependency?

File: src/structures/option.rs (hunk 1 above)
Comment (shenek) [reply src/structures/option.rs]: Ack. Reverted.

File: src/structures/option.rs (hunk 2 above)
Comment (shenek) [reply src/structures/option.rs]: dtto

File: Cargo.toml (hunk 3 above)
Comment (shenek) [reply Cargo.toml]: It is not dependent on skim during compilation. It just requires the binary. Works in the same way as fzf.

**Comment by denisidoro** [comment]
Actually, since it uses the skim binary (instead of skim library) I see no reason to use compile-time checks for this.

I think we could have an option `--finder` (which fallbacks to `$NAVI_FINDER` which fallbacks to `fzf`)

I'll merge this on Monday, ok?

**Comment by shenek** [comment]
I've done the changes required to pass `--finder` option. So feel free to check it and merge it.

**Pull request merged**